platform/search/src/services/GenericSearchProvider.js

```diff
@@ -58,6 +58,8 @@ define([
 
         this.pendingQueries = {};
 
+        this.useGenericIndexer = openmct.useGenericIndexer;
```

Comment:
As you say, this should be private, so I wonder if we should use the legacy APIs ["constants" extension point](https://github.com/nasa/openmct/blob/de8f8d174d1a4e32ea9caa816d8795cb34267e9e/example/msl/bundle.js#L69) - full documentation [here] (https://nasa.github.io/openmct/docs/guide/#configuration)? It would you to define a constant (eg. USE_LEGACY_INDEXER) as a dependency to the GenericSearchProvider. In openmct we would define it so that the new search worker is used by default. Then, in VISTA we could add a true/false option to vista-config.js that if true registers a second `USE_LEGACY_INDEXER` constant but with [a higher priority](https://nasa.github.io/openmct/docs/guide/#priority) which would override the openmct default value.

Reply:
@akhenry - Thats a good call, its cleaner and more private for 'internal' use only. I made the change